client/rest/src/main/java/org/opensearch/client/RequestOptions.java

```diff
@@ -53,18 +56,21 @@ public final class RequestOptions {
      */
     public static final RequestOptions DEFAULT = new Builder(
         Collections.emptyList(),
+        Collections.emptyMap(),
         HeapBufferedResponseConsumerFactory.DEFAULT,
         null,
         null
     ).build();
 
     private final List<Header> headers;
+    private final Map<String, String> queryParams;
```

Comment:
Please use `parameters` , this is the way `org.opensearch.client.Request` denotes query string
```suggestion
    private final Map<String, String> parameters;
```